pkg/statestore/storeadapter/migration.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storeadapter
+
+import (
+	"strings"
+
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/storage/migration"
+)
+
+func allSteps() migration.Steps {
+	return map[uint64]migration.StepFn{
+		1: epochMigration,
+	}
+}
+
+var deleteEntries = []string{
```

Comment:
we also need to nuke the addressbook cause there were changes to hive
imo it would to be easier to have a list of prefixes to preserve (exclude)
eg `[]string{"accounting", "pseudosettle", "swap", "overlayV2_nonce"}`

Reply:
> we also need to nuke the addressbook cause there were changes to hive imo it would to be easier to have a list of prefixes to preserve (exclude) eg `[]string{"accounting", "pseudosettle", "swap", "overlayV2_nonce"}`

We have a lot more entries. I went through the entries and found:
"accounting",                                                                                                       
"pseudosettle",                                                                                                     
"swap",                                                                                                             
"overlayV2_nonce",                                                                                                  
"non-mineable-overlay",                                                                                             
"root-pin",                                                                                                         
"postage",                                                                                                          
"transaction",                                                                                                      
"nonce-overlay",                                                                                                    
"batchstore",                                                                                                       
"batchservice"

I think the exclude list is smaller. I will add the addressbook entries.



> Do you think it's necessary to resync batches? It would make sure that all expired chunks are properly pruned from the localstore so that no garbage chunk gets migrated. We can also achieve this during epoch migration by calling the valid stamp func for each chunk?

If we dont migrate the batchstore entries it will do a full resync and evict anything if required. I think its nice to have but can cause problems with the RPC endpoints. Wdyt?

Reply:
I think we have to do some sort of checking before the node starts, resyncing the batches is the easiest. I think it should be ok with the rpc endpoints, the contracts are not that old.

The other solution is to iterate the reserve once and check each chunk's stamp  validation.

Reply:
Yes I agree... Lets do a resync. Is it okay to delete both the batchstore and batchservice entries?

Reply:
yes, let's delete both